www/scripts/codecheckerviewer/BugFilterView.js

```diff
@@ -1038,7 +1046,9 @@ function (declare, Deferred, dom, domClass, all, topic, Standby, Button,
           class    : 'difftype',
           title    : 'Diff type',
           parent   : this,
-          defaultValues : [CC_OBJECTS.DiffType.NEW],
+          defaultValues : function () {
+            return { [this.class] : [CC_OBJECTS.DiffType.NEW] };
```

Comment:
What is the meaning of an array as an object key? This array must have one element and that will be the key. In JavaScript the key of an object is always a string even if it is given as a number or any other type. In this specific case if `this.class` is an object then its string representation will be the key: `"[object Object]"`. I've seen this strange array key in other parts of the code too.

Reply:
Here I'm using a variable as an object key.
```js
var x = 'test';
var obj = { x : 10 }; // obj is { x : 10 }
```

```js
var x = 'test';
var obj = { [x] : 10 }; // obj is { test : 10 }
```

In my case I'm returning an object which key will be the class value:
`{ [this.class] : [CC_OBJECTS.DiffType.NEW] }` will be `{ difftype : [CC_OBJECTS.DiffType.NEW] }`